### PR TITLE
add format strings to specs

### PR DESCRIPTION
**Pull request by ben-lerch**
here's a first pass at adding format strings to the specs

### CODE REVIEW

File: pbreports/report/specs/mapping_stats.json [hunk 1]
```diff
@@ -70,84 +70,84 @@
         {
             "id": "mapped_subread_concordance_mean",
             "type": "float",
-            "format": null,
+            "format": "{:.2f}",
```
Comment (natechols) [inline pbreports/report/specs/mapping_stats.json]: Based on Marty's spreadsheet, I think this actually needs to be `{p:.2f}`

File: pbreports/report/specs/mapping_stats.json [hunk 2]
```diff
@@ -14,49 +14,49 @@
...
-                    "format": null,
+                    "format": "{:,d}",
                     "id": "mapped_subread_length",
                     "description": "The mean length of post-filter subreads that mapped to the reference sequence."
                 },
                 {
                     "header": "Mean Concordance (mapped)",
                     "type": "float",
-                    "format": null,
+                    "format": "{:.2f}",
```
Comment (natechols) [inline pbreports/report/specs/mapping_stats.json]: `{p:.2f}`, see comment below

File: pbreports/report/specs/variants.json [hunk 3]
```diff
@@ -14,28 +14,28 @@
...
-                    "format": null,
+                    "format": "{p:.2f}",
                     "id": "bases_called",
                     "description": "Percentage of the reference sequence for which consensus bases were called"
                 },
                 {
                     "header": "Consensus Concordance",
                     "type": "float",
-                    "format": null,
+                    "format": "{:.2f}",
```
Comment (natechols) [inline pbreports/report/specs/variants.json]: this should be a percentage

File: pbreports/report/specs/variants.json [hunk 4]
```diff
@@ -14,28 +14,28 @@
...
                     "id": "concordance",
                     "description": "The percent concordance of the consensus sequence compared to the reference"
                 },
                 {
                     "header": "Base Coverage",
                     "type": "float",
-                    "format": null,
+                    "format": "{:.2f}",
```
Comment (natechols) [inline pbreports/report/specs/variants.json]: also a percentage

File: pbreports/report/specs/variants.json [hunk 5]
```diff
@@ -69,14 +69,14 @@
         },
         {
             "type": "float",
-            "format": null,
+            "format": "{p:.2f}",
```
Comment (natechols) [inline pbreports/report/specs/variants.json]: see above

File: pbreports/report/specs/variants.json (hunk 4 above)
Comment (ben-lerch) [inline pbreports/report/specs/variants.json]: Does "Base Coverage" map to "Mean Coverage" in Marty's spreadsheet? This value is currently not represented as a percentage in smrtlink.

File: pbreports/report/specs/variants.json (hunk 5 above)
Comment (ben-lerch) [inline pbreports/report/specs/variants.json]: the metric here is: "Percent Reference Bases Called (mean)" so i think the formatting string is correct

Comment (natechols) [inline pbreports/report/specs/variants.json]: Sorry, you're right, I misread.  It's the one below this that's wrong.

File: pbreports/report/specs/variants.json [hunk 6]
```diff
@@ -48,14 +48,14 @@
     "attributes": [
         {
             "type": "float",
-            "format": null,
+            "format": "{:.2f}",
```
Comment (natechols) [inline pbreports/report/specs/variants.json]: This one, however, should be a percent.

File: pbreports/report/specs/variants.json (hunk 4 above)
Comment (natechols) [inline pbreports/report/specs/variants.json]: Yes, you're right.

**Comment by ben-lerch** [comment]
the reports all seem to be version "1.0.0" but i'm hunting for an exception. otherwise i think this can be merge. i wanted to get something in quickly so i'm not blocking nat.

**Comment by natechols** [comment]
I'm not sure what to do with the version now, because pbcommand outputs the *model* version, and it there's no way to override this with the *spec* version.  @mpkocher what should we be doing here?

**Comment by mpkocher** [comment]
It's not completely clear to me how to handle this. 

One possible model is where the version is the SL version which all report specific model is encapsulated under a single version string for each SL release.

**Comment by natechols** [comment]
Let's just go with mirroring the SL version for now.  Ben, please make this change ("3.2" everywhere) and we'll merge.

**Pull request merged**